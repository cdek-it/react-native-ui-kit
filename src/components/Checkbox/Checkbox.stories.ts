import type { Meta, StoryObj } from '@storybook/react'

import { Checkbox } from './Checkbox'

const meta: Meta<typeof Checkbox> = {
  title: 'Checkbox',
  component: Checkbox,
  args: {
    checked: false,
    indeterminate: false,
    disabled: false,
    state: 'default',
  },
  argTypes: { state: { control: 'radio', options: ['default', 'danger'] } },
}

export default meta

type Story = StoryObj<typeof Checkbox>

const CheckboxStory: Story = {}

export { CheckboxStory as Checkbox }
