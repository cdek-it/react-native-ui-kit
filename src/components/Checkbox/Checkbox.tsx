import { IconCheck, IconMinus } from '@tabler/icons-react-native'
import React, { memo, useMemo } from 'react'
import { type AccessibilityProps, Pressable } from 'react-native'
import type { ViewProps } from 'react-native-svg/lib/typescript/fabric/utils'

import { makeStyles } from '../../utils/makeStyles'

import { usePressableStyles } from './hooks/usePressableStyles'

type CheckboxState = 'default' | 'danger'

export interface CheckboxProps
  extends AccessibilityProps,
    Pick<ViewProps, 'testID'> {
  /** Обработчик нажатия на чекбокс */
  readonly onPress: () => void
  /**
   * true, если необходим компонент в активном состоянии
   * @default false
   */
  readonly checked?: boolean
  /**
   * true, если необходим компонент в неопределенном состоянии
   * @default false
   */
  readonly indeterminate?: boolean
  /**
   * Управление доступностью компонента
   * @default false
   */
  readonly disabled?: boolean
  /** Выбор состояния компонента */
  readonly state: CheckboxState
}

/**
 * Используется для множественного выбора элементов
 * @see https://www.figma.com/design/4TYeki0MDLhfPGJstbIicf/UI-kit-PrimeFace-(DS)?node-id=484-5316
 */
export const Checkbox = memo<CheckboxProps>(
  ({
    onPress,
    checked = false,
    disabled = false,
    indeterminate = false,
    testID,
    state,
  }: CheckboxProps) => {
    const Icon = useMemo(() => {
      if (indeterminate) {
        return IconMinus
      }

      if (checked) {
        return IconCheck
      }

      return null
    }, [indeterminate, checked])

    const pressableStyles = usePressableStyles({
      checked,
      indeterminate,
      disabled,
      state,
    })
    const styles = useStyles()

    return (
      <Pressable
        accessibilityRole='button'
        disabled={disabled}
        hitSlop={10}
        style={pressableStyles}
        testID={testID || 'CheckboxButton_Pressable'}
        onPress={onPress}
      >
        {Icon ? (
          <Icon
            height={styles.icon.height}
            style={styles.icon}
            width={styles.icon.width}
          />
        ) : null}
      </Pressable>
    )
  }
)

const useStyles = makeStyles(({ theme }) => ({
  icon: {
    height: 14,
    width: 14,
    color: theme.Form.Checkbox.checkboxIconActiveColor,
  },
}))
