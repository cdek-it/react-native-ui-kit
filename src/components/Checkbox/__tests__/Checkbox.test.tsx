import { render } from '@testing-library/react-native'

import { Checkbox, type CheckboxProps } from '../Checkbox'

describe('Checkbox', () => {
  const defaultProps: CheckboxProps = {
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    onPress: () => {},
    state: 'default',
  }

  describe('snapshots', () => {
    const snapshotCases: Array<[string, Partial<CheckboxProps>]> = [
      ['default', {}],
      [
        'checked = false, indeterminate = false, disabled = false, state = default',
        {
          checked: false,
          indeterminate: false,
          disabled: false,
          state: 'default',
        },
      ],
      [
        'checked = true, indeterminate = false, disabled = false, state = default',
        {
          checked: true,
          indeterminate: false,
          disabled: false,
          state: 'default',
        },
      ],
      [
        'checked = true, indeterminate = true, disabled = false, state = default',
        {
          checked: true,
          indeterminate: true,
          disabled: false,
          state: 'default',
        },
      ],
      [
        'checked = true, indeterminate = true, disabled = true, state = default',
        {
          checked: true,
          indeterminate: true,
          disabled: true,
          state: 'default',
        },
      ],
      [
        'checked = false, indeterminate = false, disabled = false, state = danger',
        {
          checked: false,
          indeterminate: false,
          disabled: false,
          state: 'danger',
        },
      ],
      [
        'checked = true, indeterminate = false, disabled = false, state = danger',
        {
          checked: true,
          indeterminate: false,
          disabled: false,
          state: 'danger',
        },
      ],
      [
        'checked = true, indeterminate = true, disabled = false, state = danger',
        {
          checked: true,
          indeterminate: true,
          disabled: false,
          state: 'danger',
        },
      ],
      [
        'checked = true, indeterminate = true, disabled = true, state = danger',
        { checked: true, indeterminate: true, disabled: true, state: 'danger' },
      ],
    ]

    test.each(snapshotCases)('%s', (_, props) => {
      const { toJSON } = render(<Checkbox {...defaultProps} {...props} />)

      expect(toJSON()).toMatchSnapshot()
    })
  })
})
