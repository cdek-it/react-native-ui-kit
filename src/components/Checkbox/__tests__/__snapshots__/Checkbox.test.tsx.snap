// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox snapshots checked = false, indeterminate = false, disabled = false, state = danger 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={10}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 3.5,
        "borderWidth": 1,
        "height": 21,
        "justifyContent": "center",
        "marginLeft": 3.5,
        "marginTop": 3.5,
        "width": 21,
      },
      {
        "backgroundColor": "#ffffff",
        "borderColor": "#e26e63",
        "outlineColor": "#f6d4d1",
        "outlineStyle": "solid",
        "outlineWidth": 3.5,
      },
      false,
      false,
    ]
  }
  testID="CheckboxButton_Pressable"
/>
`;

exports[`Checkbox snapshots checked = false, indeterminate = false, disabled = false, state = default 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={10}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 3.5,
        "borderWidth": 1,
        "height": 21,
        "justifyContent": "center",
        "marginLeft": 3.5,
        "marginTop": 3.5,
        "width": 21,
      },
      {
        "backgroundColor": "#ffffff",
        "borderColor": "#dbdada",
      },
      false,
      false,
    ]
  }
  testID="CheckboxButton_Pressable"
/>
`;

exports[`Checkbox snapshots checked = true, indeterminate = false, disabled = false, state = danger 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={10}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 3.5,
        "borderWidth": 1,
        "height": 21,
        "justifyContent": "center",
        "marginLeft": 3.5,
        "marginTop": 3.5,
        "width": 21,
      },
      {
        "backgroundColor": "#188700",
        "borderColor": "#e26e63",
        "outlineColor": "#f6d4d1",
        "outlineStyle": "solid",
        "outlineWidth": 3.5,
      },
      false,
      false,
    ]
  }
  testID="CheckboxButton_Pressable"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={14}
    bbWidth={14}
    fill="none"
    focusable={false}
    height={14}
    meetOrSlice={0}
    minX={0}
    minY={0}
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "color": "#ffffff",
          "height": 14,
          "width": 14,
        },
        {
          "flex": 0,
          "height": 14,
          "width": 14,
        },
      ]
    }
    vbHeight={24}
    vbWidth={24}
    width={14}
    xmlns="http://www.w3.org/2000/svg"
  >
    <RNSVGGroup
      color="#ffffff"
      fill={null}
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
          "strokeLinecap",
          "strokeLinejoin",
        ]
      }
      stroke={
        {
          "type": 2,
        }
      }
      strokeLinecap={1}
      strokeLinejoin={1}
      strokeWidth={2}
    >
      <RNSVGPath
        color="#ffffff"
        d="M5 12l5 5l10 -10"
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "type": 2,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`Checkbox snapshots checked = true, indeterminate = false, disabled = false, state = default 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={10}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 3.5,
        "borderWidth": 1,
        "height": 21,
        "justifyContent": "center",
        "marginLeft": 3.5,
        "marginTop": 3.5,
        "width": 21,
      },
      {
        "backgroundColor": "#188700",
        "borderColor": "#188700",
      },
      false,
      false,
    ]
  }
  testID="CheckboxButton_Pressable"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={14}
    bbWidth={14}
    fill="none"
    focusable={false}
    height={14}
    meetOrSlice={0}
    minX={0}
    minY={0}
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "color": "#ffffff",
          "height": 14,
          "width": 14,
        },
        {
          "flex": 0,
          "height": 14,
          "width": 14,
        },
      ]
    }
    vbHeight={24}
    vbWidth={24}
    width={14}
    xmlns="http://www.w3.org/2000/svg"
  >
    <RNSVGGroup
      color="#ffffff"
      fill={null}
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
          "strokeLinecap",
          "strokeLinejoin",
        ]
      }
      stroke={
        {
          "type": 2,
        }
      }
      strokeLinecap={1}
      strokeLinejoin={1}
      strokeWidth={2}
    >
      <RNSVGPath
        color="#ffffff"
        d="M5 12l5 5l10 -10"
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "type": 2,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`Checkbox snapshots checked = true, indeterminate = true, disabled = false, state = danger 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={10}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 3.5,
        "borderWidth": 1,
        "height": 21,
        "justifyContent": "center",
        "marginLeft": 3.5,
        "marginTop": 3.5,
        "width": 21,
      },
      {
        "backgroundColor": "#188700",
        "borderColor": "#e26e63",
        "outlineColor": "#f6d4d1",
        "outlineStyle": "solid",
        "outlineWidth": 3.5,
      },
      false,
      false,
    ]
  }
  testID="CheckboxButton_Pressable"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={14}
    bbWidth={14}
    fill="none"
    focusable={false}
    height={14}
    meetOrSlice={0}
    minX={0}
    minY={0}
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "color": "#ffffff",
          "height": 14,
          "width": 14,
        },
        {
          "flex": 0,
          "height": 14,
          "width": 14,
        },
      ]
    }
    vbHeight={24}
    vbWidth={24}
    width={14}
    xmlns="http://www.w3.org/2000/svg"
  >
    <RNSVGGroup
      color="#ffffff"
      fill={null}
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
          "strokeLinecap",
          "strokeLinejoin",
        ]
      }
      stroke={
        {
          "type": 2,
        }
      }
      strokeLinecap={1}
      strokeLinejoin={1}
      strokeWidth={2}
    >
      <RNSVGPath
        color="#ffffff"
        d="M5 12l14 0"
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "type": 2,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`Checkbox snapshots checked = true, indeterminate = true, disabled = false, state = default 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={10}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 3.5,
        "borderWidth": 1,
        "height": 21,
        "justifyContent": "center",
        "marginLeft": 3.5,
        "marginTop": 3.5,
        "width": 21,
      },
      {
        "backgroundColor": "#188700",
        "borderColor": "#188700",
      },
      false,
      false,
    ]
  }
  testID="CheckboxButton_Pressable"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={14}
    bbWidth={14}
    fill="none"
    focusable={false}
    height={14}
    meetOrSlice={0}
    minX={0}
    minY={0}
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "color": "#ffffff",
          "height": 14,
          "width": 14,
        },
        {
          "flex": 0,
          "height": 14,
          "width": 14,
        },
      ]
    }
    vbHeight={24}
    vbWidth={24}
    width={14}
    xmlns="http://www.w3.org/2000/svg"
  >
    <RNSVGGroup
      color="#ffffff"
      fill={null}
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
          "strokeLinecap",
          "strokeLinejoin",
        ]
      }
      stroke={
        {
          "type": 2,
        }
      }
      strokeLinecap={1}
      strokeLinejoin={1}
      strokeWidth={2}
    >
      <RNSVGPath
        color="#ffffff"
        d="M5 12l14 0"
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "type": 2,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`Checkbox snapshots checked = true, indeterminate = true, disabled = true, state = danger 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={10}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 3.5,
        "borderWidth": 1,
        "height": 21,
        "justifyContent": "center",
        "marginLeft": 3.5,
        "marginTop": 3.5,
        "width": 21,
      },
      {
        "backgroundColor": "#188700",
        "borderColor": "#e26e63",
        "outlineColor": "#f6d4d1",
        "outlineStyle": "solid",
        "outlineWidth": 3.5,
      },
      {
        "backgroundColor": "grey",
        "borderColor": "grey",
      },
      false,
    ]
  }
  testID="CheckboxButton_Pressable"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={14}
    bbWidth={14}
    fill="none"
    focusable={false}
    height={14}
    meetOrSlice={0}
    minX={0}
    minY={0}
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "color": "#ffffff",
          "height": 14,
          "width": 14,
        },
        {
          "flex": 0,
          "height": 14,
          "width": 14,
        },
      ]
    }
    vbHeight={24}
    vbWidth={24}
    width={14}
    xmlns="http://www.w3.org/2000/svg"
  >
    <RNSVGGroup
      color="#ffffff"
      fill={null}
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
          "strokeLinecap",
          "strokeLinejoin",
        ]
      }
      stroke={
        {
          "type": 2,
        }
      }
      strokeLinecap={1}
      strokeLinejoin={1}
      strokeWidth={2}
    >
      <RNSVGPath
        color="#ffffff"
        d="M5 12l14 0"
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "type": 2,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`Checkbox snapshots checked = true, indeterminate = true, disabled = true, state = default 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={10}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 3.5,
        "borderWidth": 1,
        "height": 21,
        "justifyContent": "center",
        "marginLeft": 3.5,
        "marginTop": 3.5,
        "width": 21,
      },
      {
        "backgroundColor": "#188700",
        "borderColor": "#188700",
      },
      {
        "backgroundColor": "grey",
        "borderColor": "grey",
      },
      false,
    ]
  }
  testID="CheckboxButton_Pressable"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={14}
    bbWidth={14}
    fill="none"
    focusable={false}
    height={14}
    meetOrSlice={0}
    minX={0}
    minY={0}
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "color": "#ffffff",
          "height": 14,
          "width": 14,
        },
        {
          "flex": 0,
          "height": 14,
          "width": 14,
        },
      ]
    }
    vbHeight={24}
    vbWidth={24}
    width={14}
    xmlns="http://www.w3.org/2000/svg"
  >
    <RNSVGGroup
      color="#ffffff"
      fill={null}
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
          "strokeLinecap",
          "strokeLinejoin",
        ]
      }
      stroke={
        {
          "type": 2,
        }
      }
      strokeLinecap={1}
      strokeLinejoin={1}
      strokeWidth={2}
    >
      <RNSVGPath
        color="#ffffff"
        d="M5 12l14 0"
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "type": 2,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`Checkbox snapshots default 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={10}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 3.5,
        "borderWidth": 1,
        "height": 21,
        "justifyContent": "center",
        "marginLeft": 3.5,
        "marginTop": 3.5,
        "width": 21,
      },
      {
        "backgroundColor": "#ffffff",
        "borderColor": "#dbdada",
      },
      false,
      false,
    ]
  }
  testID="CheckboxButton_Pressable"
/>
`;
